## Matrix Multiplication by simple method

In [18]:
import time as T
A=[[1,2,3],[4,5,6]]
B=[[1,2,3,4],[5,6,7,8],[9,10,11,12]]
ANS=[[0 for _ in range(4)] for _ in range(2)]

starting=t.time()

for i in range(2):
    for j in range(4):
         for k in range(3):
                ANS[i][j]+=A[i][k]*B[k][j]
print("Multiplied matrix is : ")
print(ANS)

ending=t.time()
taken=ending-starting
print("Time taken is : {:.10f}".format(taken))

Multiplied matrix is : 
[[38, 44, 50, 56], [83, 98, 113, 128]]
Time taken is : 0.0021543503


## Matrix multiplication using numpy

In [21]:
import time as T
import numpy as np
A=[[1,2,3],[4,5,6]]
B=[[1,2,3,4],[5,6,7,8],[9,10,11,12]]

starting=T.time()
ANS=np.dot(A,B)
print("Multiplied matrix is : ")
print(ANS)
ending=T.time()
taken=ending-starting
print("TIme taken is : {:.10f}".format(taken))



Multiplied matrix is : 
[[ 38  44  50  56]
 [ 83  98 113 128]]
TIme taken is : 0.0007243156


->Here we can see that time taken for matrix multiplication using NUMPY(0.0007243156) is lesser than the same by normal method(0.0021543503) .. So we can say that NUMPY is faster than Normal matrix Multipliacation mathod 
->This is happening because NUMPY uses vectorization means entire vector will be processed at once instead of processing each and eevery single elements 